solutions/fscloud/main.tf

```diff
@@ -0,0 +1,22 @@
+module "resource_group" {
+  count                        = (var.resource_group_id == null) ? 1 : 0
```

Comment:
You don't need a count here - this module will always run. It will create a new RG is `existing_resource_group` is `false` or it will return the RG ID if `existing_resource_group` is `true`